src/sec.c

```diff
@@ -23,11 +23,10 @@ sec_bin_init(sec_bin_t *bin) {
 bool
 sec_init(tsdn_t *tsdn, sec_t *sec, base_t *base, pai_t *fallback,
     const sec_opts_t *opts) {
-	size_t max_alloc = opts->max_alloc & PAGE_MASK;
-	pszind_t npsizes = sz_psz2ind(max_alloc);
-	if (sz_pind2sz(npsizes) > opts->max_alloc) {
-		npsizes--;
-	}
+	assert((opts->max_alloc & PAGE_MASK) == 0);
+	assert(opts->max_alloc > 0);
+
+	pszind_t npsizes = sz_psz2ind(opts->max_alloc) + 1;
```

Comment:
Maybe another variant of the same question -- I think we should decrement this again if the (possibly-not-a-page-multiple) max_alloc is smaller than its quantization?

Reply:
Right, since all `size` inputs of `sec_alloc` will match size classes, if `max_alloc` falls in between two size classes, we should 'truncate' it to the previous size class.